backend/cmd/headlamp.go

```diff
@@ -510,16 +513,45 @@ func createHeadlampHandler(config *HeadlampConfig) http.Handler {
 		}
 		if !isURLContainedInProxyURLs {
 			http.Error(w, "no allowed proxy url match, request denied ", http.StatusBadRequest)
+			return
+		}
+
+		proxyReq, err := http.NewRequestWithContext(ctx, r.Method, proxyURL, r.Body)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		// We may want to filter some headers, otherwise we could just use a shallow copy
+		// proxyReq.Header = req.Header
+		proxyReq.Header = make(http.Header)
+		for h, val := range r.Header {
+			proxyReq.Header[h] = val
+		}
+
+		// Disable caching
```

Comment:
I'm wondering if disabling caching should be up to the client? If the client wants caching stuff to be cached, then this wouldn't allow that.

Reply:
This is fine to improve later IMO.

Reply:
I'm not so sure. Previously we didn't disable caching, so this could perhaps break something that relied on it.

Reply:
@ashu8912  Can you expand on the issues that you faced with the external proxy that led to disabling cache?

Reply:
So while proxying to get the chart detail  the /externalproxy endpoint was cached, So even if i was on a different detail view i would get data from the last cached /externalproxy

Reply:
From Ashu's description, we should NOT be caching in this endpoint. Insteatd setting those cache headers in the client. Because clients are responsible for setting client side caching headers.

It seems this code has been moved out of this PR, so I'll resolve this convo.